src/main.js

```diff
@@ -98,11 +98,17 @@ module.exports = {
         }
         const filePath = textEditor.getPath()
         const showRule = atom.config.get('linter-eslint.showRuleIdInMessage')
+        let rules = {}
+
+        if (textEditor.isModified()) {
+          rules = idsToIgnoredRules(atom.config.get('linter-eslint.silenceWhileTypingRules'))
```

Comment:
Observe the setting in `activate` so it only has to do the complex parsing when things change.